### PR TITLE
fix(components): ConfirmationModal accepts markdown

### PR DESCRIPTION
**Pull request by glowatsk**
## Motivations

This repo was linked in the Dev Edmonton slack, was looking for opportunities for Hacktoberfest.

I chose to make the Markdown component accept an optional value over the confirmation modal requiring a message. I believe this maintains better feature parity and will allow Markdown to be used in more optional components in the future.

### Added

- Added markdown support to the confirmation modal
- Allowed an optional value in markdown to support blank modals
- Update modal snapshot

### Changed

- 

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

- Current tests pass
- Can add markdown to the generated page
---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

### CODE REVIEW

File: packages/components/src/ConfirmationModal/ConfirmationModal.mdx [hunk 1]
```diff
@@ -33,7 +33,7 @@ import {
         <Button label="Open" onClick={() => setOpen(true)} />
         <ConfirmationModal
           title="Should we?"
-          message="Do something…"
+          message={"### Do something..."}
```
Comment (eddysims) [inline packages/components/src/ConfirmationModal/ConfirmationModal.mdx:36]: good example here. As a note, we try to keep the examples to a more realistic approach. In the case of `ConfirmationModal`, we would not likely have a heading in the message. Something like `message={`Let's **do something**`}` may be more realistic.

File: packages/components/src/Markdown/Markdown.tsx [hunk 2]
```diff
@@ -9,7 +9,7 @@ interface MarkdownProps {
   /**
    * Rich text content.
    */
-  readonly content: string;
+  readonly content?: string;
```
Comment (eddysims) [inline packages/components/src/Markdown/Markdown.tsx:12]: Making this prop optional would allow `Markdown` to be used without any content. That shouldn't ever be a use case. for the markdown component.

File: packages/components/src/ConfirmationModal/ConfirmationModal.tsx [hunk 3]
```diff
@@ -220,7 +255,11 @@ export const ConfirmationModal = forwardRef(function ConfirmationModalInternal(
       }}
     >
       <Content>
-        <Text>{state.message}</Text>
+        <Markdown
+          content={state.message}
+          externalLink={state.externalLink}
+          basicUsage={state.basicUsage}
```
Comment (eddysims) [inline packages/components/src/ConfirmationModal/ConfirmationModal.tsx:261]: I think that these props are out of scope for https://github.com/GetJobber/atlantis/issues/381. In the issue, the only change should be that the message prop can accept markdown.

**Review by eddysims — CHANGES_REQUESTED** [review]
@glowatsk Thanks for this! A few comments for you to look over.

File: packages/components/src/Markdown/Markdown.tsx (hunk 2 above)
Comment (glowatsk) [reply packages/components/src/Markdown/Markdown.tsx:12]: How would you handle an optional message prop in the Confirmation Modal with a required content prop?

File: packages/components/src/ConfirmationModal/ConfirmationModal.mdx (hunk 1 above)
Comment (glowatsk) [reply packages/components/src/ConfirmationModal/ConfirmationModal.mdx:36]: Good change, I will update it.

File: packages/components/src/ConfirmationModal/ConfirmationModal.tsx (hunk 3 above)
Comment (glowatsk) [reply packages/components/src/ConfirmationModal/ConfirmationModal.tsx:261]: The issue requested using the Markdown component, should it not be possible to pass these props to the markdown component?

**Review by eddysims — CHANGES_REQUESTED** [review]

File: packages/components/src/ConfirmationModal/ConfirmationModal.tsx (hunk 3 above)
Comment (glowatsk) [reply packages/components/src/ConfirmationModal/ConfirmationModal.tsx]: Removed!

File: packages/components/src/Markdown/Markdown.tsx (hunk 2 above)
Comment (glowatsk) [reply packages/components/src/Markdown/Markdown.tsx]: Added!

**Review by glowatsk — COMMENTED** [review]
Updated @eddysims

**Comment by chris-at-jobber** [comment]
@glowatsk we've fixed the issue causing specs to fail - if you update this branch from the latest `master` we can get a review going for you 😄

**Comment by glowatsk** [comment]
@chris-at-jobber I have re-based the branch, seems like everything is still working as intended.

**Comment by eddysims** [comment]
@glowatsk You are probably not able to see the failure. Heres what it is saying

```
Error: atlantis/packages/components/src/ConfirmationModal/ConfirmationModal.tsx(224,21): semantic error TS2322 Type 'string | undefined' is not assignable to type 'string'.
  Type 'undefined' is not assignable to type 'string'.
```

**Comment by glowatsk** [comment]
@eddysims thank you, fixed.

**Comment by eddysims** [comment]
@glowatsk looks like there are some issues with https://deploy-preview-389--jobber-atlantis.netlify.app/components/confirmation-modal#controlled-confirmation-modal

I know that Hacktoberfest is over. If you would like I can take this from here and push it over. Let me know if you would like us to take over the task.

**Comment by glowatsk** [comment]
Sure go ahead and take it over.

**Comment by eddysims** [comment]
Closing in favor of https://github.com/GetJobber/atlantis/pull/424
